### ISSUE TITLE
downloadExpoGoAsync - TypeError: Cannot read properties of null (reading 'update')

### ISSUE THREAD

**Issue by joelpierre** [expo/expo#21672]
### Summary

Within the `downloadExpoGoAsync` onProgress function for a clean android emulator install I got the following error:
```
TypeError: Cannot read properties of null (reading 'update')
```

This is because of the below:

```
let bar = null;
...
onProgress ({ progress , total  }) {
                if (progress && total) {
                    if (!bar) {
                        if (spinner.isSpinning) {
                            spinner.stop();
                        }
                        bar = (0, _progress).createProgressBar("Downloading the Expo Go app [:bar] :percent :etas", {
                            width: 64,
                            total: 100,
                            // clear: true,
                            complete: "=",
                            incomplete: " "
                        });
                    }
                    bar.update(progress, total); 👈🏽 --- this is the problem.
                }
            }
```
it is assumed that `bar` will be reassigned however it isn't and blows up. I fixed it locally by doing the following 

```
if (bar) {
    bar.update(progress, total);
}
```

Not sure why it does this on Android only and didn't impact the iOS fresh install.

### What platform(s) does this occur on?

Android

### SDK Version

48.0.6

### Environment

expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 12.6.3
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 18.12.1 - ~/.nvm/versions/node/v18.12.1/bin/node
      Yarn: 3.4.1 - ~/.nvm/versions/node/v18.12.1/bin/yarn
      npm: 8.19.2 - ~/.nvm/versions/node/v18.12.1/bin/npm
      Watchman: 2023.01.23.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    IDEs:
      Android Studio: 2021.3 AI-213.7172.25.2113.9123335
      Xcode: 14.2/14C18 - /usr/bin/xcodebuild
    Expo Workflow: managed

### Minimal reproducible example

Create a new empty expo project and a new unused android emulator and run `expo start -c --android`

**Comment by pixelpax** [comment]
Experiencing this as well. I'm experiencing it on MacOS with an ios build

**Comment by maks-io** [comment]
I had the same problem **but with ios**, after updating xcode on my system. The solution was to manually run `expo client:install:ios` once, afterwards i was able to start up my app again. I'm assuming it could be fixed similarly on android.

**Comment by megancooper** [comment]
> I had the same problem **but with ios**, after updating xcode on my system. The solution was to manually run `expo client:install:ios` once, afterwards i was able to start up my app again. I'm assuming it could be fixed similarly on android.

This also worked for me, but I had to add the `expo-cli` package first (for anyone else that's new to expo like I am 😅 ):

```bash
$ pnpm add -g expo-cli
$ expo client:install:ios
```

**Comment by SeanDemske** [comment]
Experienced the same issue as OP on Windows targeting Android. Using the null check fix, running the command, and then reverting the file is our current work around and has fixed subsequent attempts

**Comment by 0618** [comment]
I got the same error on macOS.

I tried `npm i -g expo-cli` then`expo client:install:ios`, but got the following error:
> $ expo client:install:ios is not supported in the local CLI, please use npx expo start --ios instead

And still seeing

> node_modules/@expo/cli/build/src/utils/downloadExpoGoAsync.js:103
                    bar.update(progress, total);
                        ^

TypeError: Cannot read properties of null (reading 'update')

**Comment by 0618** [comment]
I saw another person reported this issue on Discord: https://discord.com/channels/695411232856997968/943205994123657246/1084082722013728768

**Comment by 0618** [comment]
Even though I made the change in https://github.com/expo/expo/pull/22398, I'm still having the same problem on Github Workflow :(

**Comment by brentvatne** [comment]
please do not use `expo start` - this is the deprecated global cli. use `npx expo start`, which will invoke the local cli from the `expo` package. this is the recommendation for sdk 46+ - https://blog.expo.dev/the-new-expo-cli-f4250d8e3421

**Comment by brentvatne** [comment]
just looked at the linked pr - it looks like this is indeed reported with the local cli after all

**Comment by brentvatne** [comment]
i published the fix in `expo@48.0.19`
